core/common/junit/src/main/java/org/eclipse/edc/junit/extensions/EmbeddedRuntime.java

```diff
@@ -70,54 +71,51 @@ private EmbeddedRuntime(MultiSourceServiceLocator serviceLocator, String name, M
     @Override
     public void boot(boolean addShutdownHook) {
         var monitor = super.createMonitor();
-        try {
-            monitor.info("Starting runtime %s".formatted(name));
 
-            // Temporarily inject system properties.
-            var savedProperties = (Properties) System.getProperties().clone();
-            properties.forEach(System::setProperty);
+        monitor.info("Starting runtime %s".formatted(name));
 
-            var runtimeException = new AtomicReference<Exception>();
-            var latch = new CountDownLatch(1);
+        serviceLocator.registerSystemExtension(ConfigurationExtension.class, (ConfigurationExtension) () -> ConfigFactory.fromMap(properties));
 
-            runtimeThread = executorService.submit(() -> {
-                try {
-                    var classLoader = URLClassLoader.newInstance(classPathEntries);
+        var runtimeThrowable = new AtomicReference<Throwable>();
+        var latch = new CountDownLatch(1);
 
-                    Thread.currentThread().setContextClassLoader(classLoader);
+        runtimeThread = executorService.submit(() -> {
+            try {
+                var classLoader = URLClassLoader.newInstance(classPathEntries);
 
-                    super.boot(false);
+                Thread.currentThread().setContextClassLoader(classLoader);
```

Comment:
I just saw this now. Technically this should reset the previous classloader for the TCCL in a finally block (on error) or when the runtime shuts down. Probably for another PR.

Reply:
It shouldn't be necessary as this code runs in a separated thread, so this class assignment is done only for the specific thread

Reply:
it's currently working like this:
https://github.com/eclipse-edc/Connector/blob/c530bcf29a86e5b1ed1218eda5ec2c75bc4bc838/core/common/junit/src/main/java/org/eclipse/edc/junit/extensions/EmbeddedRuntime.java#L87